### PR TITLE
split miscellaneous code into a new command_helpers module

### PR DESCRIPTION
**Pull request by Be-ing**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3823,82 +3826,6 @@ impl Tool {
...
-        };
-
-        objects.push(Object::new(file.to_path_buf(), obj));
-    }
-
-    Ok(objects)
-}
-
 #[cfg(feature = "parallel")]
 fn try_wait_on_child(
```
[the remark this answers is not in the document]
Comment (Be-ing) [reply src/lib.rs]: done

File: src/lib.rs [hunk 2]
```diff
@@ -1424,7 +1427,7 @@ impl Build {
                     }
                 };
                 let pipe_writer = print.map(PrintThread::clone_pipe_writer).transpose()?;
-                let child = spawn(&mut cmd, &program, pipe_writer)?;
+                let child = command_helpers::spawn(&mut cmd, &program, pipe_writer)?;
```
[the remark this answers is not in the document]
Comment (NobodyXu) [reply src/lib.rs:1438]: Isn't there a `use command_helper::*;` alteady?

**Review by NobodyXu — CHANGES_REQUESTED** [review]
BTW, it seems that you copied `Build` into src/build.rs, however there's no `mod build;` declaration anywhere in the diff.

File: src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Be-ing) [reply src/lib.rs]: Ah yes, there is. Fixed.

**Comment by Be-ing** [comment]
> BTW, it seems that you copied Build into src/build.rs, however there's no mod build; declaration anywhere in the diff.

Oops, I didn't mean to commit that in this PR. I am waiting for this, #924, and #929 to be merged before doing that.

**Review by NobodyXu — APPROVED** [review]
Thanks!

**Pull request merged**